modules/snooker/src/main/ts/ephox/snooker/lookup/Type.ts

```diff
@@ -39,7 +39,7 @@ const getRowHeaderType = (isHeaderRow: boolean, isHeaderCells: boolean): RowHead
 const getRowType = (row: CommonRowDetails): RowDetails => {
   // Header rows can use a combination of theads and ths - want to detect the different combinations
   const isHeaderRow = row.section === 'thead';
-  const isHeaderCells = Optionals.is(findCommonCellType(row.cells), 'th');
+  const isHeaderCells = row.section === 'tbody' && Optionals.is(findCommonCellType(row.cells), 'th');
```

Comment:
The failing test is because now `th` inside a `thead` are not being set as header cells.

I suggest something like this

```suggestion
  const isBodyRow = row.section === 'tbody';
  const isHeaderCells = (isHeaderRow || isBodyRow) && Optionals.is(findCommonCellType(row.cells), 'th');
```

Reply:
Is that correct behavior though?

Reply:
The suggest James left looks correct to me... as anything can be a header as long as it's in the `thead` or `tbody`. That said, I do have to wonder if the other suggestion I left on the Jira would be better. If we move `if (row.section === 'tfoot') {` before the other checks then it'll show the `tfoot` check has precedence :shrug: